### PR TITLE
Support specifying the XDP action(s) as CLI args

### PR DESCRIPTION
**Pull request by arthurfabre**
Add support for specifying the desired XDP actions(s) as CLI args. Filter programs will only be loaded / attached for those actions, capturing only those packets.

This also reworks the flag parsing / handling first so we can add some tests.

### CODE REVIEW

File: cmd/xdpcap/flags.go [hunk 1]
```diff
@@ -4,11 +4,53 @@ import (
...
 	"strings"
 
 	"github.com/pkg/errors"
 )
 
+type actionsFlag []xdpAction
+
+func (a *actionsFlag) Set(val string) error {
+	// Clear defaults
+	*a = (*a)[:0]
```
Comment (lmb) [inline cmd/xdpcap/flags.go]: Just use a new slice?

File: cmd/xdpcap/flags.go [hunk 2]
```diff
@@ -4,11 +4,53 @@ import (
...
+		}
+	}
+
+	return nil
+}
+
+func (a *actionsFlag) String() string {
+	strs := []string{}
+	for _, action := range *a {
+		strs = append(strs, action.String())
```
Comment (lmb) [inline cmd/xdpcap/flags.go]: This isn't going to produce the input, but something like `XDPPass,XDPDrop` no?

File: cmd/xdpcap/flags.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// nopWriter implements io.Writer and discards everything written to it
+type nopWriter struct {
```
Comment (lmb) [inline cmd/xdpcap/flags.go]: Check out [ioutil.Discard](https://golang.org/pkg/io/ioutil/#pkg-variables).

File: cmd/xdpcap/flags.go [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+Capture packets from XDP programs matching a tcpdump filter expression.
+
+<output> may be "-" to write to stdout. Implies -q and -flush.
+
+`, flags.name))
+
+	// temporarily re-enable printing to get the default usage
+	flags.FlagSet.SetOutput(&usage)
+	flags.PrintDefaults()
```
Comment (lmb) [inline cmd/xdpcap/flags.go]: Why not `.FlagSet` here?

File: cmd/xdpcap/flags.go [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+// nopWriter implements io.Writer and discards everything written to it
+type nopWriter struct {
+}
+
+func (nopWriter) Write(p []byte) (int, error) {
+	return len(p), nil
+}
+
+type flags struct {
+	*flag.FlagSet
```
Comment (lmb) [inline cmd/xdpcap/flags.go]: Split this into `config struct{}` and a free function that uses a FlagSet instead? I'm not a fan of mixing the source of config with the config itself.

File: cmd/xdpcap/flags.go [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+	// temporarily re-enable printing to get the default usage
+	flags.FlagSet.SetOutput(&usage)
+	flags.PrintDefaults()
+	flags.FlagSet.SetOutput(nopWriter{})
+
+	return usage.String()
+}
+
+func (flags *flags) Parse(args []string) (*captureOpts, error) {
```
Comment (lmb) [inline cmd/xdpcap/flags.go]: This should return `config` of some sort

File: cmd/xdpcap/flags.go (hunk 1 above)
Comment (arthurfabre) [reply cmd/xdpcap/flags.go]: Done :)

File: cmd/xdpcap/flags.go (hunk 2 above)
Comment (arthurfabre) [reply cmd/xdpcap/flags.go]: I used stringer with `-stripprefix`, so it did print `Pass`, `Drop`.. . This wasn't very clear though, and `stringer` doesn't support parsing stuff, so I've gotten rid of string and manually created the `xdpAction.String()` and `parseAction()` functions.

File: cmd/xdpcap/flags.go (hunk 4 above)
Comment (arthurfabre) [reply cmd/xdpcap/flags.go]: Good point, removed `.FlagSet` everywhere for consistency.

File: cmd/xdpcap/flags.go (hunk 5 above)
Comment (arthurfabre) [reply cmd/xdpcap/flags.go]: Merged the `captureOpts` and `flags` types, and `newFlags()` and `flags.Parse` into `parseFlags()`, which I think is clearer.

File: cmd/xdpcap/flags.go (hunk 6 above)
Comment (arthurfabre) [reply cmd/xdpcap/flags.go]: `flags.Parse` has been with `newFlags()` into `parseFlags()`, which just returns a `flags`.

File: cmd/xdpcap/flags.go [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+
+	quiet bool
+	flush bool
+
+	filterExpr string
+	filterOpts filterOpts
+}
+
+// parseFlags creates the flags, and attempts to parse args.
+// Returned flags will never be nil, even on error.
```
Comment (lmb) [inline cmd/xdpcap/flags.go]: Nit: It's not a pointer, so it can't be nil?

Comment (arthurfabre) [reply cmd/xdpcap/flags.go]: Oops, yes. Fixed

**Pull request merged**